34-cellular-modbus-client/README.md

```diff
@@ -0,0 +1,427 @@
+# Cellular Modbus Client
+
+Send requests and receive responses from Modbus servers via cellular.
+
+- [Cellular Modbus Client](#cellular-modbus-client)
+  - [You Will Need](#you-will-need)
+  - [Hardware Setup](#hardware-setup)
+    - [RS-485 Transceiver Breakout](#rs-485-transceiver-breakout)
+    - [USB to RS-485 Converter](#usb-to-rs-485-converter)
+    - [Notecarrier](#notecarrier)
+  - [Notehub Setup](#notehub-setup)
+  - [Firmware](#firmware)
+    - [Operation](#operation)
+        - [Request Types](#request-types)
+            - [Read Coils](#read-coils)
+            - [Read Discrete Inputs](#read-discrete-inputs)
+            - [Read Holding Registers](#read-holding-registers)
+            - [Read Input Registers](#read-input-registers)
+            - [Write Single Coil](#write-single-coil)
+            - [Write Single Register](#write-single-register)
+            - [Write Multiple Coils](#write-multiple-coils)
+            - [Write Multiple Registers](#write-multiple-registers)
+        - [Response Types](#response-types)
+            - [Reads](#reads)
+            - [Writes](#writes)
+            - [Errors](#errors)
+  - [Testing](#testing)
+    - [Simple Test](#simple-test)
+    - [Test Script](#test-script)
+  - [Blues Community](#blues-community)
+  - [Additional Resources](#additional-resources)
+
+## You Will Need
+
+* [Visual Studio Code (VS Code)](https://code.visualstudio.com/) with the [PlatformIO extension](https://platformio.org/)
+* [Notecarrier F](https://blues.io/products/notecarrier/notecarrier-f/)
+* [Notecard](https://blues.io/products/notecard/)
+* [SparkFun Transceiver Breakout - RS-485](https://www.sparkfun.com/products/10124)
+* [SparkFun USB to RS-485 Converter](https://www.sparkfun.com/products/9822)
+* [2 Screw Terminals 3.5mm Pitch (3-Pin)](https://www.sparkfun.com/products/8235)
+* [STLINK-V3MINI](https://shop.blues.io/collections/accessories/products/stlink-v3mini)
+* 2 micro USB cables
+* [1 USB mini-B cable](https://www.sparkfun.com/products/13243)
+* Male to male jumper wires
+* Female to male jumper wires
+
+## Hardware Setup
+
+This application uses [Modbus RTU](https://en.wikipedia.org/wiki/Modbus#Modbus_RTU_frame_format) with RS-485 for Modbus communication between the client and server. The RS-485 transceiver breakout is used by the client running on the Swan to send and receive Modbus messages. The USB to RS-485 converter is used by the server running on your development PC to send and receive Modbus messages.
+
+### RS-485 Transceiver Breakout
+
+To use the RS-485 transceiver breakout board, solder on a row of 5 male headers and one of the 3-pin screw terminals:
+
+![RS-485 transceiver breakout soldered](images/rs485_transceiver_breakout_soldered.jpg)
```

Comment:
Very trivial but you could trim this image a bit so it’s not so tall.

Reply:
Yeah, I'll crop it. I'm going to make all the images smaller, too. They're huge.